src/python/pants/core/goals/publish.py

```diff
@@ -320,8 +328,10 @@ async def package_for_publish(request: PublishProcessesRequest) -> PublishProces
     publish = await MultiGet(
         Get(
             PublishProcesses,
-            PublishRequest,
-            field_set._request(packages),
+            {
+                field_set._request(packages): PublishRequest,
+                local_environment.val: EnvironmentName,
+            },
```

Comment:
@stuhood This is where the publish processes are constructed, pinned to the local environment.

Reply:
Yea, you're right! I forgot that `PublishProcesses` doesn't actually do any of the building, and that happens via `BuiltPackage`.